components/message_submit_error.tsx

```diff
@@ -2,19 +2,29 @@
 // See LICENSE.txt for license information.
 
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, {ReactFragment} from 'react';
 import {FormattedMessage} from 'react-intl';
 
 import {isErrorInvalidSlashCommand} from 'utils/post_utils.jsx';
 
-class MessageSubmitError extends React.PureComponent {
-    static propTypes = {
+interface ErrorMessage {
+    message: string | ReactFragment;
+}
+
+interface MessageSubmitErrorProps {
+    error: ErrorMessage;
+    handleSubmit: (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+    submittedMessage: string;
+}
+
+class MessageSubmitError extends React.PureComponent<MessageSubmitErrorProps, {}> {
+    public static propTypes = {
```

Comment:
This isn't needed now that we're using TypeScript

Reply:
@devinbinnie thanks for the review. I wanted to take them out earlier but did some googling and found this answer `https://stackoverflow.com/a/54690878/7723621`. Basically TypeScript validates at compilation time and proptypes validate at runtime. What do you think of the answer?

Reply:
They effectively do the same thing in this case, since we're not bringing data in from an API or an external source or anything like that. Every use is internal to our app. 

I don't think it's necessary to have both, and we've been removing them in other migrations. So for consistency, I say we remove them.

Reply:
@devinbinnie thanks for the review :) . I have made the change.